spki/src/traits.rs

```diff
@@ -93,3 +96,11 @@ pub trait EncodePublicKey {
         Ok(doc.write_pem_file(path, SubjectPublicKeyInfoRef::PEM_LABEL, line_ending)?)
     }
 }
+
+/// Returns AlgorithmIndentifier associated with the structure. This is mostly useful for Signing
+/// and Validation keys.
+#[cfg(feature = "alloc")]
+pub trait AssociatedAlgorithmIdentifier {
+    /// Returns corresponding AlgorithmIndentifier
+    fn get_algo_id(&self) -> AlgorithmIdentifierOwned;
```

Comment:
Also while I'm here, how about this name for this method:

```suggestion
    fn algorithm_identifier(&self) -> AlgorithmIdentifierOwned;
```